### PR TITLE
Feat/theme ui theming

### PR DESCRIPTION
**Pull request by jonathan-fielding**
Add theme ui based theming based on Eevee work done by Recharge, this adds theming to the button only however its straightforward to apply to other components later,

### CODE REVIEW

File: src/elements/button/src/stories.tsx [hunk 1]
```diff
@@ -2,77 +2,34 @@
...
 import { Icon } from '../../icon/src'
 
 import { Button } from './'
 
 const Spacer = () => <div css={css({ minHeight: 20 })} />
 
-storiesOf('Elements|Button', module).add('primary variant', () => (
+storiesOf('Elements|Button', module).addDecorator(Story => {
+  return <ThemeProvider theme={theme()}>{Story()}</ThemeProvider>
+}).add('primary variant', () => (
```
Comment (brendonboshell) [inline src/elements/button/src/stories.tsx]: Could this decorator be defined in `.storybook/config.js` rather than on specific component stories?

**Comment by brendonboshell** [comment]
@jonathan-fielding Does `npx lerna run build` work for you? I get the following error:

```
src/index.tsx(4,34): error TS7016: Could not find a declaration file for module 'theme-ui'. '/Users/brendonboshell/webdev/trustyle/src/elements/button/node_modules/theme-ui/dist/index.js' implicitly has an 'any' type.
  Try `npm install @types/theme-ui` if it exists or add a new declaration (.d.ts) file containing `declare module 'theme-ui';`
src/index.tsx(21,6): error TS2322: Type '{ children: ReactNode; href?: string | undefined; autoFocus?: boolean | undefined; form?: string | undefined; formAction?: string | undefined; formEncType?: string | undefined; formMethod?: string | undefined; ... 258 more ...; onClick: ((event: MouseEvent<...>) => void) | undefined; }' is not assignable to type 'DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>'.
  Property 'sx' does not exist on type 'DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>'.
```

Once I install `@types/theme-ui`, I get the following:

```
src/index.tsx(23,12): error TS2532: Object is possibly 'undefined'.
src/index.tsx(23,26): error TS2339: Property 'base' does not exist on type 'ObjectOrArray<StandardProperties<string | 0>>'.
  Property 'base' does not exist on type 'StandardProperties<string | 0>[]'.
```

File: src/elements/button/src/stories.tsx (hunk 1 above)
Comment (jonathan-fielding) [reply src/elements/button/src/stories.tsx]: Unfortunately not, it simply didnt work as it lost the context of the theme in the component

**Comment by brendonboshell** [comment]
I am a little bit confused with how the styled-system theme works, here are the types (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/styled-system/index.d.ts#L223):

```
export interface Theme {
    breakpoints?: string[] | number[] | object;
    mediaQueries?: { [size: string]: string };
    space?: ObjectOrArray<number | string>;
    fontSizes?: ObjectOrArray<CSS.FontSizeProperty<number>>;
    colors?: ObjectOrArray<CSS.ColorProperty>;
    fonts?: ObjectOrArray<CSS.FontFamilyProperty>;
    fontWeights?: ObjectOrArray<CSS.FontWeightProperty>;
    lineHeights?: ObjectOrArray<CSS.LineHeightProperty<{}>>;
    letterSpacings?: ObjectOrArray<CSS.LetterSpacingProperty<{}>>;
    sizes?: ObjectOrArray<CSS.HeightProperty<{}> | CSS.WidthProperty<{}>>;
    borders?: ObjectOrArray<CSS.BorderProperty<{}>>;
    borderStyles?: ObjectOrArray<CSS.BorderProperty<{}>>;
    borderWidths?: ObjectOrArray<CSS.BorderWidthProperty<{}>>;
    radii?: ObjectOrArray<CSS.BorderRadiusProperty<{}>>;
    shadows?: ObjectOrArray<CSS.BoxShadowProperty>;
    zIndices?: ObjectOrArray<CSS.ZIndexProperty>;
    buttons?: ObjectOrArray<CSS.StandardProperties>;
    colorStyles?: ObjectOrArray<CSS.StandardProperties>;
    textStyles?: ObjectOrArray<CSS.StandardProperties>;
}
```

This seems incompatible with the themes we are defining, e.g. we have a `theme.card` prop. The `buttons` seem to expect CSS props rather than `theme.buttons.primary`.

Do we need to override the types?

**Comment by jonathan-fielding** [comment]
We fixed typing with @brendonboshell

**Pull request merged**